plugins/inputs/prometheus/prometheus.go

```diff
@@ -216,16 +217,15 @@ func (p *Prometheus) GetAllURLs() (map[string]URLAndAddress, error) {
 // Reads stats from all configured servers accumulates stats.
 // Returns one of the errors encountered while gather stats (if any).
 func (p *Prometheus) Gather(acc telegraf.Accumulator) error {
-	if p.client == nil {
-		client, err := p.createHTTPClient()
-		if err != nil {
-			return err
-		}
-		p.client = client
-		p.headers = map[string]string{
-			"User-Agent": internal.ProductToken(),
-			"Accept":     acceptHeader,
-		}
+	ctx := context.Background()
+	client, err := p.HTTPClientConfig.CreateClient(ctx, p.Log)
+	if err != nil {
+		return err
+	}
+	p.client = client
+	p.headers = map[string]string{
+		"User-Agent": internal.ProductToken(),
+		"Accept":     acceptHeader,
 	}
```

Comment:
Please move this to an `Init() error` function as it isn't really _required_ to recreate the client in every `Gather()` cycle... The `Init()` function is then called automatically by Telegraf after setting all user-configuration-options.